File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {  
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        if(root == null){
            return root == subRoot;
        }

        return sameTree(root, subRoot) || isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot);
    }

    public boolean sameTree(TreeNode currNode, TreeNode currSubNode){
        if(currSubNode == null && currNode == null){
            return true;
        } else if(currNode == null || currSubNode == null){
            return false;
        }

        return currNode.val == currSubNode.val && sameTree(currNode.left, currSubNode.left) && sameTree(currNode.right, currSubNode.right);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        queue = collections.deque()
        queue.append(root)
        res = []

        while queue:
            level = []
            size = len(queue)
            for i in range(size):
                curr_node = queue.popleft()
                level.append(curr_node.val)
                if curr_node.left:
                    queue.append(curr_node.left)
                if curr_node.right:
                    queue.append(curr_node.right)
            res.append(level)


        return res